src/grumpkin/interop_tests.rs

```diff
@@ -0,0 +1,197 @@
+use ark_ec::{AffineCurve, ProjectiveCurve};
+use ark_ff::{BigInteger256, PrimeField, Zero};
+use ark_serialize::CanonicalSerialize;
+use grumpkin::{Fq, SWAffine, SWProjective};
+// use std::convert::TryFrom;
+
+#[test]
+fn c_plus_plus_interop_generator() {
+    // Copied from the C++ codebase
+    let expected_x = "0000000000000000000000000000000000000000000000000000000000000001";
+    let expected_y = "0000000000000002cf135e7506a45d632d270d45f1181294833fc48d823f272c";
+
+    let gen = SWAffine::prime_subgroup_generator();
+
+    assert!(gen.is_on_curve());
+    assert!(gen.is_in_correct_subgroup_assuming_on_curve());
+
+    let mut bytes_x = Vec::new();
+    let mut bytes_y = Vec::new();
+    gen.x.serialize(&mut bytes_x).unwrap();
+    gen.y.serialize(&mut bytes_y).unwrap();
+
+    bytes_x.reverse();
+    bytes_y.reverse();
+
+    assert_eq!(hex::encode(bytes_x), expected_x);
+    assert_eq!(hex::encode(bytes_y), expected_y);
+}
+
+#[test]
+fn c_interop_derive_generators() {
+    let generators = get_generators_string();
+
+    let new_generators = generators.replace(&['{', '}', ',', '\n'], "");
+    let g_list: Vec<&str> = new_generators.split_whitespace().collect();
+
+    for affine_elem in g_list.chunks(2) {
+        let x_bytes = hex::decode(&affine_elem[0][2..]).unwrap();
+        let y_bytes = hex::decode(&affine_elem[1][2..]).unwrap();
+        let point = deserialise_point(&x_bytes, &y_bytes).unwrap();
+        assert!(point.is_on_curve());
+        assert!(point.is_in_correct_subgroup_assuming_on_curve());
+    }
+}
+
+fn deserialise_fq(bytes: &[u8]) -> Option<Fq> {
+    assert_eq!(bytes.len(), 32);
+
+    let mut tmp = BigInteger256([0, 0, 0, 0]);
+
+    tmp.0[3] = u64::from_be_bytes(<[u8; 8]>::try_from(&bytes[0..8]).unwrap());
+    tmp.0[2] = u64::from_be_bytes(<[u8; 8]>::try_from(&bytes[8..16]).unwrap());
+    tmp.0[1] = u64::from_be_bytes(<[u8; 8]>::try_from(&bytes[16..24]).unwrap());
+    tmp.0[0] = u64::from_be_bytes(<[u8; 8]>::try_from(&bytes[24..32]).unwrap());
+
+    Fq::from_repr(tmp)
+}
+
+fn deserialise_point(x_bytes: &[u8], y_bytes: &[u8]) -> Option<SWAffine> {
+    let x = deserialise_fq(x_bytes)?;
+    let y = deserialise_fq(y_bytes)?;
+    let is_infinity = false; // none of the generators should be points at infinity
+    Some(SWAffine::new(x, y, is_infinity))
+}
+
+fn get_generators_string() -> &'static str {
+    let generators: &str = "{ 0x01d1774edd499b0f18a8f4c641577596ab54a7ef5c26a6e4f2576c434af2a09f, 0x0756cf7a70bcc863e9995227f9cd576a3ddf9bcd239f154da3e2dd1a3c63e762 }
```

Comment:
We can make this a constant outside of the function instead:

const GENERATORS : [[&str;2];128] = [
   [ "0x0a32482e544b1275f7c2c794f6aec98f3877aa5df828912ed932902a2563ab19",    "0x0aa5cb6f338a6352f1390630e755d0173cfed3a24b01dd93c06069a45473afd7"],
   
   ...

]

Since rust doesn't like 0x prefixes, we can just grep to remove it